schema/rpc.fbs

```diff
@@ -560,3 +563,112 @@ table SaveFileNotification {
     /// Recommended filename
     expected_filename: string;
 }
+
+// ------------ Firmware tool -------------
+
+enum FlashingMethod: uint8 {
+    NONE = 0,
+    OTA,
+    SERIAL
+}
+
+enum FirmwareUpdateStatus: uint8 {
+    /// The server is downloading the firmware
+    DOWNLOADING,
+
+    /// The server is waiting for the tracker to be rebooted by the user
+    /// Note that is is not the same as REBOOTING
+    WAITING_FOR_REBOOT,
+
+    /// The server tries to authenticate with the MCU
+    AUTHENTICATING,
+
+    /// The server is uploading the firmware to the Device
+    UPLOADING,
+
+    /// The serial flasher tries to sync with the MCU
+    /// You can use this event to prompt the user to press the boot btn
+    SYNCING_WITH_MCU,
+
+    /// The MCU is rebooting
+    REBOOTING,
+
+    /// The server is privisioning the tracker
+    PROVISIONING,
+
+    // The update process completed with success
+    DONE,
+
+    /// Could not find the device
+    ERROR_DEVICE_NOT_FOUND,
+
+    /// The operation timed out, > 1min
+    ERROR_TIMEOUT,
+
+    /// The firmware download failed
+    ERROR_DOWNLOAD_FAILED,
+
+    /// The server could not authenticate with the MCU
+    ERROR_AUTHENTICATION_FAILED,
+
+    /// Could not upload the firmware to the MUC
+    ERROR_UPLOAD_FAILED,
+
+    /// The provision of the tracker failed, usually wifi credentials
+    ERROR_PROVISIONING_FAILED,
+
+    /// An unsupported Flashing method was used
+    ERROR_UNSUPPORTED_METHOD,
+
+    ERROR_UNKNOWN,
+}
+
+table SerialDeviceId {
```

Comment:
Naming scheme here is a bit weird. Usually IDs are monotonically increasing integers, assigned by the server - effectively, they function as pointers. But this appears to be a wrapper type/type alias for a Serial port (like COM3). Can you clarify your intention here?